firmware/applications/SignBuddy/nucleo_board.h

```diff
@@ -50,60 +50,129 @@
-#define FLEX_RING_CHANNEL         LL_ADC_CHANNEL_7
-#define FLEX_LITTLE_PORT          GPIOB
-#define FLEX_LITTLE_PIN           LL_GPIO_PIN_0
-#define FLEX_LITTLE_CHANNEL       LL_ADC_CHANNEL_8
-#define FLEX_GPIO_CLK_EN()        LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOA); \
+#define FLEX_THUMB_PORT             GPIOA
+#define FLEX_THUMB_PIN              LL_GPIO_PIN_0
+#define FLEX_THUMB_CHANNEL          LL_ADC_CHANNEL_0
+#define FLEX_INDEX_PORT             GPIOA
+#define FLEX_INDEX_PIN              LL_GPIO_PIN_1
+#define FLEX_INDEX_CHANNEL          LL_ADC_CHANNEL_1
+#define FLEX_MIDDLE_PORT            GPIOA
+#define FLEX_MIDDLE_PIN             LL_GPIO_PIN_6
+#define FLEX_MIDDLE_CHANNEL         LL_ADC_CHANNEL_6
+#define FLEX_RING_PORT              GPIOA
+#define FLEX_RING_PIN               LL_GPIO_PIN_7
+#define FLEX_RING_CHANNEL           LL_ADC_CHANNEL_7
+#define FLEX_LITTLE_PORT            GPIOB
+#define FLEX_LITTLE_PIN             LL_GPIO_PIN_0
+#define FLEX_LITTLE_CHANNEL         LL_ADC_CHANNEL_8
+#define FLEX_GPIO_CLK_EN()          LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOA); \
   LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOB);
 
 /* I2C */
-#define IMU_I2C                   I2C1
-#define IMU_I2C_SCL_PIN           LL_GPIO_PIN_6
-#define IMU_I2C_SDA_PIN           LL_GPIO_PIN_7
-#define IMU_I2C_GPIO_AF           LL_GPIO_AF_1
-#define IMU_I2C_GPIO_PORT         GPIOB
-#define IMU_I2C_CLK_SRC()         LL_RCC_SetI2CClockSource(LL_RCC_I2C1_CLKSOURCE_PCLK1);
-#define IMU_I2C_CLK_EN()          LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOB);
-#define IMU_I2C_GPIO_CLK_EN()     LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_I2C1);
+#define IMU_I2C                     I2C1
+#define IMU_I2C_SCL_PIN             LL_GPIO_PIN_6
+#define IMU_I2C_SDA_PIN             LL_GPIO_PIN_7
+#define IMU_I2C_GPIO_AF             LL_GPIO_AF_1
+#define IMU_I2C_GPIO_PORT           GPIOB
+#define IMU_I2C_CLK_SRC()           LL_RCC_SetI2CClockSource(LL_RCC_I2C1_CLKSOURCE_PCLK1);
+#define IMU_I2C_CLK_EN()            LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOB);
+#define IMU_I2C_GPIO_CLK_EN()       LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_I2C1);
 
 /* TSC */
-#define TSC_ELECTRODE_PORT        GPIOC
-#define TSC_ELECTRODE_PIN         LL_GPIO_PIN_5
-#define TSC_ELECTRODE_AF          LL_GPIO_AF_3
-#define TSC_ELECTRODE_IO          TSC_GROUP3_IO1
-
-#define TSC_SAMPLER_PORT          GPIOB
-#define TSC_SAMPLER_PIN           LL_GPIO_PIN_2
-#define TSC_SAMPLER_AF            LL_GPIO_AF_3
-#define TSC_SAMPLER_IO            TSC_GROUP3_IO4
-
-#define TSC_CLK_EN()              LL_AHB1_GRP1_EnableClock(LL_AHB1_GRP1_PERIPH_TSC)
-#define TSC_GPIO_CLK_EN()         LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOB); \
+#define TSC_ELECTRODE_AF            LL_GPIO_AF_3
+#define TSC_SAMPLER_AF              LL_GPIO_AF_3
+
+// GROUP 3
+#define TSC_GROUP3_CHANNEL0_PORT    GPIOC
+#define TSC_GROUP3_CHANNEL0_PIN     LL_GPIO_PIN_5
+#define TSC_GROUP3_CHANNEL0_IO      TSC_GROUP3_IO1
```

Comment:
We could name these TOUCH_1_PORT, TOUCH1_PIN, etc I think that would be more readable

Reply:
I can't remember the exact reason I chose this over what you're saying, but I think I had one. Let me go try and change it to that and we'll see if there's any reason we can't

Reply:
right, so the reason I did it like this is since the way we're sampling them is very close to the hardware, and dependent on which channel and which group you are using. 

I think we'd run into trouble if define TOUCH_1_PORT, TOUCH1_PIN because if we were to change something, we then have to go look at which group it is in and it would be a mess in my opinion.

I think we should leave it like this and then add:


```
#define TOUCH_SENSOR_K TOUCH_SENSOR_1
#define TOUCH_SENSOR_M TOUCH_SENSOR
... etc
```

Reply:
I see what you're saving, yeah this is fine with me